Create CONTRIBUTING.md (#42)
olin/studentgovernment: CONTRIBUTING.md

# How to Contribute

So, you've found something you'd like to change about the Student Government Bylaws or Amendments. You've come to the right place!

Any questions? Unsure of how to proceed? Reach out to any members of Student Government's Executive Board (the President, Vice President for Finance, and Vice President of Communications).

> As of the 2017-2018 school year, those members are [Sam Myers](https://github.com/sammyers), [Ana Krishnan](https://github.com/askrishnan), and [Evan New-Schmidt](https://github.com/newsch).

Reviewer: If we're going to update this, we should probably change it to the current years' folks. I also think it might be smart to change it to mailto links to make it more accessible to folks and also because there is no great way to private message someone on github.
```suggestion
As of the 2019-2020 school year, those members are [Anusha Datar](mailto:anusha.datar@students.olin.edu), [Kyle Emmi](mailto:kyle.emmi@students.olin.edu), and [Sarah Deng](mailto:sarah.deng@students.olin.edu).
```

## General Formatting

### Wrap Lines at 80 Characters

Lines should be wrapped at 80 characters - this is a fairly common standard for code and plaintext files, dating back to the days of [punchcards](https://softwareengineering.stackexchange.com/questions/148677/why-is-80-characters-the-standard-limit-for-code-width#148678).

Here's a good prime number that's eighty digits long:  

> `18532395500947174450709383384936679868383424444311405679463280782405796233163977`

Reviewer: if we want to include this, we should probably add some context for why it's here

> `18532395500947174450709383384936679868383424444311405679463280782405796233163977`

Reviewer: If we want to include the bit with the prime number, we should probably provide some context - use this as a header to align stuff? etc.

Reviewer: Change here looks good to me, resolving comment.

### Pull Request Titles

Pull request titles should follow the format:  
**[BA/CA]**-**[school year]**-**[amendment number]**: **[Summary of changes]**

> - amendment numbers are iterated for each proposal regardless of success

Reviewer: It might be nice to include a link here for what bylaw we're on. Or we could make it straightforward and say:
```suggestion
- Amendment numbers are iterated for each proposal regardless of success. Check the [history of previous PRs](https://github.com/olin/studentgovernment/pulls?utf8=%E2%9C%93&q=is%3Apr) to determine what the next number is.
```

Author: > It might be nice to include a link here for what bylaw we're on.

Love it! I'll just clean up the url a bit.

> There's also definitely space here for a switch to zero indexing :)

While that does sit better with me, for consistency's sake and our non-programmer friends, I'd recommend keeping it 1-indexed.

> - amendment numbers are counted independently for the Bylaws and Constitution

Reviewer: Might be good to add links here in case of confusions about which is which:
```suggestion
- amendment numbers are counted independently for the [Bylaws](https://github.com/olin/studentgovernment/blob/master/bylaws.md) and [Constitution](https://github.com/olin/studentgovernment/blob/master/constitution.md)
```

Author: I switched those to relative links, otherwise like the idea.

For example:

> - the first bylaw amendment of the 17-18 school year was about updating CCO procedures and was named:  

Reviewer: Might be nice to capitalize, we don't have to. 
```suggestion
- The first bylaw amendment of the 17-18 school year was about updating CCO procedures and was named:  
```

Author: Consistency is always nice!

>   `BA-1718-001: CCO by-laws update to reflect current practices`

Reviewer: Might be nice to add a link for further clarification of how the PR ought to look.
```suggestion
  [`BA-1718-001: CCO by-laws update to reflect current practices`](https://github.com/olin/studentgovernment/pull/30)
```

### Logging Changes in the README

A summary of changes should be added to the [README](README.md) in the appropriate location:
- Bylaw Amendments: [Student Government By-Laws>Passed Amendments](https://github.com/olin/studentgovernment#passed-amendments)
- Student Government Constitution: [Student Government Constitution Constitution>Passed Amendments](https://github.com/olin/studentgovernment#student-government-constitution).

> ### Referring to Positions

Reviewer: What is this section for?

Author: Mmm, I think my plan was to have some tips for how to phrase amendments. There's a lot more that could be talked about there, and a link to the location in the constitution/bylaws would probably be helpful.

Author: I've dropped that section.

Reviewer: I think that makes sense for now.

Student Government positions should be referred to as follows:

- the President
- the Vice President for Finance
- the Vice President of Communications

## Versioning the governing documents

After approving amendments in a session of student government and merging the appropriate pull requests into the `master` branch, the repository should be "tagged" with a new version, and a release made on GitHub that covers the changes.

> `vX.Y.Z`, where X is major overhauls of the documents, Y is modifications that
> impact how SG operates, and Z is changes that don't impact how student
> government operates.

More information on the history of this can be found in [issue #40](https://github.com/olin/studentgovernment/issues/40).

You can read more about tagging commits and making releases at the [GitHub documentation](https://help.github.com/en/articles/creating-releases).

> ## Testing format

Reviewer: @newsch Do you have more insight on what you were thinking of including here? We can also omit it if you don't think it's necessary to include. Same goes for the generating PDFs portion.

Author: This was one of the more brainstorm-y parts. I was thinking this would be a good place to write some of the meta processes, like converting the bylaws to pdfs and running the markdown linters. I'll write something up for you to take a look at.

Reviewer: Awesome, thank you so much!
